### PR TITLE
CLOUDP-57047: set up an E2E test with cloud manager

### PR DESCRIPTION
**Pull request by gssbzn**

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -37,13 +37,26 @@ steps:
...
     image: golang:1.13
     commands:
-      - make build
+      - make e2e-test
     volumes:
       - name: deps
         path: /go
+    environment:
+      MCLI_PUBLIC_API_KEY:
+        from_secret: mcli_public_api_key
```
Comment (colm-quinn) [inline .drone.yml]: [q] Did this end up targeting cloud dev/QA in the end?

Comment (gssbzn) [reply .drone.yml]: had a chat with @jamesbroadhead today we are now doing against dev

File: .drone.yml [hunk 2]
```diff
@@ -37,13 +37,26 @@ steps:
...
+      MCLI_PRIVATE_API_KEY:
+        from_secret: mcli_private_api_key
+      MCLI_OPS_MANAGER_URL:
+        from_secret: mcli_ops_manager_url
```
Comment (colm-quinn) [inline .drone.yml]: [q] How does it setup/get the ops manager instance?

File: e2e/atlas_clusters_test.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	atlasEntity := "atlas"
+	clustersEntity := "clusters"
+	clusterName := fmt.Sprintf("e2e-cluster-%v", r.Uint32())
+
+	t.Run("Create", func(t *testing.T) {
```
Comment (colm-quinn) [inline e2e/atlas_clusters_test.go]: [q] Is it possible for us to define command keywords centrally somewhere, so they can be reused here and the cli definitions?

File: .drone.yml (hunk 2 above)
Comment (gssbzn) [reply .drone.yml]: talked offline

**Pull request merged**